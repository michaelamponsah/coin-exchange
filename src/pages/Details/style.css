section {
  min-height: 100vh;
}

.details-icon {
  margin: 1em;
}

.img-wrapper {
  display: flex;
  justify-content: center;
}

.detail-img {
  width: 80px;
  height: 80px;
  margin: 1em 0;
}

.detail-list {
  display: block;
}

.detail-list li {
  display: grid;
  grid-template-columns: 1fr 1fr;
  column-gap: 40px;
  margin: 5px 0;
  background-color: inherit;
  padding: 0.8em;
  color: #fefefe;
}

.detail-list li:nth-child(odd) {
  background-color: #072a4b;
}

span {
  display: inline-block;
}