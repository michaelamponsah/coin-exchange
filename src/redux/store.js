import thunk from 'redux-thunk';
import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import exchangesReducer from './exchange';

const store = configureStore(
  {
    reducer: {
      exchange: exchangesReducer,
    },
  },
  applyMiddleware((thunk)),
);

export default store;
