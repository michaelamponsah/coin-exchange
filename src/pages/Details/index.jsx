import React from 'react';
import { FiArrowLeftCircle } from 'react-icons/fi';
import { useSelector } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import './style.css';

const Details = () => {
  const store = useSelector((state) => state.exchange);
  const { exchanges } = store;
  const { id } = useParams();

  const data = exchanges.filter((exchange) => exchange.id === id);

  if (data.length > 0) {
    window.localStorage.setItem('details', JSON.stringify(data[0]));
  }

  const items = JSON.parse(window.localStorage.getItem('details'));

  const {
    name,
    image,
    url,
    year_established: year,
    trade_volume_24h_btc: volume,
    trust_score: trust,
    trust_score_rank: rank,
    has_trading_incentive: hasIncentive,
    country,
  } = items;

  return (
    <section>
      <div className="head">
        <NavLink to="/">
          <FiArrowLeftCircle className="icon details-icon" />
        </NavLink>
      </div>
      <div className="img-wrapper">
        <img className="detail-img" src={image} alt={name} />
      </div>
      <ul className="detail-list">
        <li>
          <span>Name: </span>
          <span>{name}</span>
        </li>
        <li>
          <span>URL: </span>
          <span>{url}</span>
        </li>
        <li>
          <span>Year established: </span>
          <span>{year}</span>
        </li>
        <li>
          <span>Country: </span>
          <span>{country}</span>
        </li>
        <li>
          <span>Trade Volume in 24 hrs (BTC): </span>
          <span>{volume}</span>
        </li>
        <li>
          <span>Has trading incentive: </span>
          <span>{hasIncentive ? 'True' : 'False'}</span>
        </li>
        <li>
          <span>Trust score: </span>
          <span>{trust}</span>
        </li>
        <li>
          <span>Trust score rank: </span>
          <span>{rank}</span>
        </li>
      </ul>
    </section>
  );
};

export default Details;
