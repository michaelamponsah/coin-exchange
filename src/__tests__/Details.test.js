import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Details from '../pages/Details';
import store from '../redux/store';

it('Details component renders as expected', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Details />
      </BrowserRouter>
    </Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
