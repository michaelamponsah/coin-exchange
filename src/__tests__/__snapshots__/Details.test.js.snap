// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders as expected 1`] = `
<section>
  <div
    className="head"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <svg
        className="icon details-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polyline
          points="12 8 8 12 12 16"
        />
        <line
          x1="16"
          x2="8"
          y1="12"
          y2="12"
        />
      </svg>
    </a>
  </div>
  <div
    className="img-wrapper"
  >
    <img
      alt="Coinbase Exchange"
      className="detail-img"
      src="https://assets.coingecko.com/markets/images/23/small/Coinbase_Coin_Primary.png?1621471875"
    />
  </div>
  <ul
    className="detail-list"
  >
    <li>
      <span>
        Name: 
      </span>
      <span>
        Coinbase Exchange
      </span>
    </li>
    <li>
      <span>
        URL: 
      </span>
      <span>
        https://www.coinbase.com
      </span>
    </li>
    <li>
      <span>
        Year established: 
      </span>
      <span>
        2012
      </span>
    </li>
    <li>
      <span>
        Country: 
      </span>
      <span>
        United States
      </span>
    </li>
    <li>
      <span>
        Trade Volume in 24 hrs (BTC): 
      </span>
      <span>
        75483.15345614518
      </span>
    </li>
    <li>
      <span>
        Has trading incentive: 
      </span>
      <span>
        False
      </span>
    </li>
    <li>
      <span>
        Trust score: 
      </span>
      <span>
        10
      </span>
    </li>
    <li>
      <span>
        Trust score rank: 
      </span>
      <span>
        1
      </span>
    </li>
  </ul>
</section>
`;
